backend/sites/Dockerfile

```diff
@@ -1,39 +1,27 @@
-# RedHat UBI 8 with nodejs 16
-FROM node:16.14.0-alpine
-
-# Set npm cache directory
-ENV NPM_CONFIG_CACHE=/home/node/.npm
-
-# NPM Permission Fix
-RUN mkdir -p /home/node/.npm \
-    && chown -R 1001:0 /home/node/.npm
-
-# Install packages, build, and keep only prod packages
+# Build
+FROM node:22-slim AS build
+# Install packages, build and keep only prod packages
+USER root
 WORKDIR /app
-#test
-
-# Copy package.json and package-lock.json
-COPY package*.json ./
-
-# Install app dependencies using npm ci
-RUN npm ci
-
-# Bundle app source
-COPY . ./
-
-# Change ownership of files to non-root user
-RUN chown -R 1001:0 /app \
-    && chmod -R g=u /app
-
-# Switch to non-root user
-USER 1001
-#test
-
-# Build the application
-RUN npm run build
+COPY *.json ./
+COPY ./src /app/src
+RUN npm ci --omit=dev && \
+    npm run build
+
+# Deployment container
+# Deploy using minimal Distroless image
+FROM gcr.io/distroless/nodejs22-debian12:nonroot
+# Set node to production
+ENV NODE_ENV=production
+
+# Copy over app
+WORKDIR /app
+COPY --from=build /app/node_modules ./node_modules
+COPY --from=build /app/dist ./dist
 
-# Expose port
-EXPOSE 4007
+# Port and health check
+EXPOSE 3000
```

Comment:
I think this is going to conflict with the frontend port on our development devices. What would be a good way to mitigate this? (Or is there some docker networking magic that takes care of the issue?)

Reply:
I see, the code is hardcoded to 4007, did not realise that, I will make it a env var, and then in local, you can change ports to avoid conflict

Reply:
Let me double check that. I know the frontend currently runs on 3000, but I don't actually know where that number comes from.